web/html/src/components/input/form-utils.ts

```diff
@@ -76,6 +78,7 @@ export function unflattenModel<T>(flatModel: Record<string, T>): TreeLikeModel<T
 
   Object.keys(flatModel)
     .sort()
+    .filter(name => !stripBlankValues || ((flatModel[name] || "") !== ""))
```

Comment:
`(flatModel[name] || "") !== ""` will not work as intended when the value is a number or a boolean.  

```ts
(0 || "") !== ""; // false
(1 || "") !== ""; // true
(false || "") !== ""; // false
(true || "") !== ""; // true
```

Similar to the suggestions in `InputBase`, I would recommend being explicit about what to do with what input instead.

Reply:
ACK